src/LibChorus/VcsDrivers/Mercurial/HgResumeTransport.cs

```diff
@@ -710,14 +710,10 @@ public bool Pull(string[] baseRevisions)
 				}
 				if (response.Status == PullStatus.Fail)
 				{
-					// Not sure that we need to abort the attempts just because .Net web request says so.  See Push also. CP 2012-06
-					errorMessage = "Pull data chunk failed";
-					_progress.WriteError(errorMessage);
```

Comment:
From the log where we saw this:

    Warning: Pull data chunk failed: Error: ConnectFailure (interrupted)
    Error: Pull data chunk failed

`PullStatus.Fail` is the default return of `PullOneChunk`; it is returned for some instances of `HttmStatusCode.BadRequest`, for an unrecognized server response, or if a `WebException` is thrown.  That's all I know.